From 5f88283d359868ae5cf806c92b80f086fe88bd9f Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:39:03 -0400
Subject: [alsa] hda: Add the pin fixup for HP Envy TS bass speaker

Message-id: <140972823573-391-patchset-jkysela@redhat.com>
Patchwork-id: 91087
O-Subject: [RHEL 7.1 PATCH 390/442] ALSA: hda - Add the pin fixup for HP Envy TS bass speaker
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

NID 0x10 seems corresponding to the bass speaker.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 8695a003a5f4f5bc88b915e1c4a56d954f810f6e)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_sigmatel.c b/sound/pci/hda/patch_sigmatel.c
index 52bdbdc..946ae40 100644
--- a/sound/pci/hda/patch_sigmatel.c
+++ b/sound/pci/hda/patch_sigmatel.c
@@ -103,6 +103,7 @@ enum {
  STAC_92HD83XXX_HP,
  STAC_HP_ENVY_BASS,
  STAC_HP_BNB13_EQ,
+ STAC_HP_ENVY_TS_BASS,
  STAC_92HD83XXX_MODELS
 };
 
@@ -2668,6 +2669,13 @@ static const struct hda_fixup stac92hd83xxx_fixups[] = {
   .chained = true,
   .chain_id = STAC_92HD83XXX_HP_MIC_LED,
  },
+ [STAC_HP_ENVY_TS_BASS] = {
+  .type = HDA_FIXUP_PINS,
+  .v.pins = (const struct hda_pintbl[]) {
+   { 0x10, 0x92170111 },
+   {}
+  },
+ },
 };
 
 static const struct hda_model_fixup stac92hd83xxx_models[] = {
@@ -2684,6 +2692,7 @@ static const struct hda_model_fixup stac92hd83xxx_models[] = {
  { .id = STAC_92HD83XXX_HEADSET_JACK, .name = "headset-jack" },
  { .id = STAC_HP_ENVY_BASS, .name = "hp-envy-bass" },
  { .id = STAC_HP_BNB13_EQ, .name = "hp-bnb13-eq" },
+ { .id = STAC_HP_ENVY_TS_BASS, .name = "hp-envy-ts-bass" },
  {}
 };
 
@@ -2739,6 +2748,8 @@ static const struct snd_pci_quirk stac92hd83xxx_fixup_tbl[] = {
      "HP bNB13", STAC_HP_BNB13_EQ),
  SND_PCI_QUIRK(PCI_VENDOR_ID_HP, 0x190A,
      "HP bNB13", STAC_HP_BNB13_EQ),
+ SND_PCI_QUIRK(PCI_VENDOR_ID_HP, 0x190e,
+     "HP ENVY TS", STAC_HP_ENVY_TS_BASS),
  SND_PCI_QUIRK(PCI_VENDOR_ID_HP, 0x1940,
      "HP bNB13", STAC_HP_BNB13_EQ),
  SND_PCI_QUIRK(PCI_VENDOR_ID_HP, 0x1941,
-- 
1.7.1